### PR TITLE
Feat 987 automatic documentation generation 

### PR DESCRIPTION
**Pull request by srieding**

### CODE REVIEW

File: src/models/translations.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ const schema = Joi.object().keys({
   en: Joi.string().allow('').example('English string').required(),
   nl: Joi.string().allow('').example('Nederlandse string').required(),
 })
-  .tag('translations');
+  .tag('translations').meta({ className: 'translations' });
```
Comment (parkerjohn) [inline src/models/translations.ts:7]: Linter should have put this on 2 lines?

File: src/models/string-or-translations.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { schema as translationsSchema, Translations } from './translations';
 const schema = Joi.alternatives(
   Joi.string().example('untranslated string'),
   translationsSchema,
-).tag('stringOrTranslations');
+).tag('stringOrTranslations').meta({ className: 'stringOrTranslations' });
```
Comment (parkerjohn) [inline src/models/string-or-translations.ts:7]: Linter should have put this on 2 lines?

File: documentation/index.ts [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+        }
+      });
+    }
+  });
+  return convertedSchemaSpecifications;
+}
+
+function createGeneralInformationMd(): string {
+  return `
+# Getting started
```
Comment (parkerjohn) [inline documentation/index.ts:197]: Maybe we could move this to another file?

File: documentation/index.ts [hunk 4]
```diff
@@ -0,0 +1,497 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import j2s, { ComponentsSchema } from 'joi-to-swagger';
+import { Schema } from 'joi';
+import path from 'path';
```
Comment (parkerjohn) [inline documentation/index.ts:4]: I get a typescript error here:

TS1259: Module '"path"' can only be default-imported using the 'esModuleInterop' flag  path.d.ts(167, 5): This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

File: documentation/index.ts [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+import fs from 'fs';
```
Comment (parkerjohn) [inline documentation/index.ts:5]: Maybe I have different typescript options set than you do?

TS1192: Module '"fs"' has no default export.

**Review by parkerjohn — COMMENTED** [review]
Do the tests run for you? I get 2 failed, 1 passed, 3 total

Is it possible to change the page title from 'ReDoc' to 'WithTheGrid SDK' or similar?

File: documentation/index.ts [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+    );
+  });
+
+  return pathSpecificationParts.join(',');
+}
+
+function determineWiddershinsOptions() {
+  return {
+    codeSamples: true,
+    httpsnippet: false,
```
Comment (parkerjohn) [inline documentation/index.ts:201]: camelCase httpSnippet

File: src/models/translations.ts (hunk 1 above)
Comment (srieding) [reply src/models/translations.ts]: fixed

File: src/models/string-or-translations.ts (hunk 2 above)
Comment (srieding) [reply src/models/string-or-translations.ts]: fixed

File: documentation/index.ts (hunk 3 above)
Comment (srieding) [reply documentation/index.ts]: fixed

File: documentation/index.ts (hunk 4 above)
Comment (srieding) [reply documentation/index.ts:4]: I don't have one anymore. You?

File: documentation/index.ts (hunk 5 above)
Comment (srieding) [reply documentation/index.ts:5]: works for me

File: documentation/index.ts (hunk 6 above)
Comment (srieding) [reply documentation/index.ts:221]: fixed

File: documentation/index.ts (hunk 4 above)
Comment (parkerjohn) [reply documentation/index.ts:4]: Yes, fixed

**Pull request merged**